code/search/FullTextSearch.php

```diff
@@ -18,16 +18,41 @@ class FullTextSearch {
 	 * @param bool $rebuild - If true, don't use cached values
 	 */
 	static function get_indexes($class = null, $rebuild = false) {
-		if ($rebuild) { self::$all_indexes = null; self::$indexes_by_subclass = array(); }
+		if ($rebuild) {
+			self::$all_indexes = null;
+			self::$indexes_by_subclass = array();
+		}
 
 		if (!$class) {
 			if (self::$all_indexes === null) {
 				$classes = ClassInfo::subclassesFor('SearchIndex');
 
-				$concrete = array();
+				$hidden = array();
+				$candidates = array();
 				foreach ($classes as $class) {
+					// Check if this index is disabled
+					$hides = $class::config()->hide_ancestor;
```

Comment:
I would prefer to make this an explicit registration system, less magic and more versatile. The YAML configuration doesn't work because we can't overwrite lists, but we could use `Config::update()` for that?

```yml
FullTextSearch:
  indexes:
    - MySolrIndex
```